UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.UnitTests/Validation/D365PayloadValidatorTest.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System.Linq;
+using FakeItEasy;
+using FluentValidation.TestHelper;
+using Microsoft.Extensions.Options;
+using NUnit.Framework;
+using UKHO.ExternalNotificationService.API.Validation;
+using UKHO.ExternalNotificationService.Common.Configuration;
+using UKHO.ExternalNotificationService.Common.Models.Request;
+
+namespace UKHO.ExternalNotificationService.API.UnitTests.Validation
+{
+    public class D365PayloadValidatorTest
+    {
+        private D365Payload _fakeD365Payload;
+        private IOptions<D365PayloadKeyConfiguration> _fakeD365PayloadKeyConfiguration;
+        private D365PayloadValidator _d365PayloadValidator;
+
+        [SetUp]
+        public void Setup()
+        {
+            _fakeD365PayloadKeyConfiguration = A.Fake<IOptions<D365PayloadKeyConfiguration>>();
+            _fakeD365PayloadKeyConfiguration.Value.PostEntityImageKey = "SubscriptionImage";
+            _fakeD365PayloadKeyConfiguration.Value.IsActiveKey = "statecode";
+            _fakeD365PayloadKeyConfiguration.Value.NotificationTypeKey = "ukho_subscriptiontype";
+            _fakeD365PayloadKeyConfiguration.Value.SubscriptionIdKey = "ukho_externalnotificationid";
+            _fakeD365PayloadKeyConfiguration.Value.WebhookUrlKey = "ukho_webhookurl";
+            _fakeD365Payload = GetD365PayloadDetails();
+
+            _d365PayloadValidator = new D365PayloadValidator(_fakeD365PayloadKeyConfiguration);
+        }
+
+        # region CorrelationId
+        [Test]
+        public void WhenNullCorrelationIdRequest_ThenRecieveBadRequest()
```

Comment:
Please correct spelling of Receive here and in other places too please:
```suggestion
        public void WhenNullCorrelationIdRequest_ThenReceiveBadRequest()
```